cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -337,13 +347,13 @@ foldMInstallPlanDepOrder plan0 visit =
 
     go !results (pkg : pkgs) = do
       -- we go in the right order so the results map has entries for all deps
-      let depresults :: ComponentDeps [b]
+      let depresults :: [b]
           depresults =
-            fmap (map (\ipkgid -> let Just result = Map.lookup ipkgid results
-                                   in result))
-                 (depends pkg)
+            map (\ipkgid -> let Just result = Map.lookup ipkgid results
+                              in result)
+                (nodeNeighbors pkg)
       result <- visit pkg depresults
-      let results' = Map.insert (installedPackageId pkg) result results
+      let results' = Map.insert (nodeKey pkg) result results
```

Comment:
Well, we could (1) newtype a DepGraph module which uses dependency graph terminology, (2) just type synonym it, (3) just create a new type class with the names you "want", which implies IsNode. Which of these do you want? (By the way, any fix here I am NOT going to rebase down the commit chain, because the commit "Be more careful about ComponentId versus UnitId." mucks around with all this business.